flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/DeletesOnKeyPrograms.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.stream;
+
+import org.apache.flink.table.test.program.SinkTestStep;
+import org.apache.flink.table.test.program.SourceTestStep;
+import org.apache.flink.table.test.program.TableTestProgram;
+import org.apache.flink.types.Row;
+import org.apache.flink.types.RowKind;
+
+/**
+ * Tests for verifying semantic of operations when sources produce deletes by key only and the sink
+ * can accept deletes by key only as well.
+ */
+public final class DeletesOnKeyPrograms {
+
+    /**
+     * Tests a simple INSERT INTO SELECT scenario where ChangelogNormalize can be eliminated since
+     * we don't need UPDATE_BEFORE, and we have key information for all changes.
+     */
+    // no ChangeLogNormalize
+    public static final TableTestProgram INSERT_SELECT_DELETE_ON_KEY_DELETE_ON_KEY =
+            TableTestProgram.of(
+                            "select-delete-on-key-to-delete-on-key",
+                            "validates results when querying source with deletes on key"
```

Comment:
```suggestion
    // no ChangeLogNormalize
    public static final TableTestProgram INSERT_SELECT_DELETE_ON_KEY_DELETE_ON_KEY =
            TableTestProgram.of(
                            "select-delete-on-key-to-delete-on-key",
                            "No ChangeLogNormalize: validates results when querying source with deletes on key"
```
How about including this into description instead of a comment?